#! /bin/sh /usr/share/dpatch/dpatch-run
## Prevent crash on opj_destroy().dpatch by  <robin.cornelius@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix a crash path in openjpeg where image->comps is wild and gets used 
## DP: in opj_destroy()

@DPATCH@

--- openjpeg-1.3+dfsg.orig/libopenjpeg/image.c	2008-03-10 08:50:35.000000000 +0000
+++ openjpeg/libopenjpeg/image.c	2009-08-16 14:11:49.000000000 +0100
@@ -36,6 +36,8 @@
 	opj_image_t *image = NULL;
 
 	image = (opj_image_t*) opj_calloc(1, sizeof(opj_image_t));
+	image->comps = NULL;
+
 	if(image) {
 		image->color_space = clrspc;
 		image->numcomps = numcmpts;
